### PR TITLE
Add tutorial for downloading, reading, writing and plotting catalog

### PR DESCRIPTION
**Pull request by seisman**
Closes https://github.com/seismo-learn/seismology101/issues/549.

### CODE REVIEW

**Comment by seisman** [comment]
@core-man I've finished the draft version for this tutorial. In this tutorial, I give three examples for `client.get_events()` and also show how to print and plot the returned `Catalog` object. At the end of the tutorial, I show how to save the catalog using `cat.write()`.

In the next section, I plan to explain how to know and use the content of Catalog and Event.

Please give this PR a first round of review before I start to polish the sentences and fix some reference issues.

**Comment by core-man** [comment]
Looks great~

File: source/_config.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ copyright : "2020–2022"
 # Force re-execution of notebooks on each build.
 # See https://jupyterbook.org/content/execute.html
 execute:
-  execute_notebooks: auto
+  execute_notebooks: force
```
Comment (seisman) [inline source/_config.yml:11]: Revert the changes before merging.
```suggestion
  execute_notebooks: auto
```

File: source/exercises/catalog.md [hunk 2]
```diff
@@ -94,12 +103,307 @@ USGS 网站提供了在线搜索和查看地震目录的功能，既可以手动
...
+print(cat)
+```
+
+## 深入理解和使用 {class}`~obspy.core.event.Catalog` 类
+
+上面提到，{meth}`Client.get_events() <obspy.clients.fdsn.client.Client.get_events()>`
+和 {func}`read_events() <obspy.core.event.read_events>` 的返回值都是
+{class}`~obspy.core.event.Catalog` 类型。
+
+事实上，{class}`~obspy.core.event.Catalog` 类是 ObsPy 中最核心的类之一，其用于储存
```
Comment (core-man) [inline source/exercises/catalog.md:231]:
```suggestion
事实上，{class}`~obspy.core.event.Catalog` 类是 ObsPy 中最核心的类之一，用于储存
```

File: source/exercises/catalog.md [hunk 3]
```diff
@@ -94,12 +103,307 @@ USGS 网站提供了在线搜索和查看地震目录的功能，既可以手动
...
+ax.set_ylabel("Counts")
+plt.show()
+```
+
 ### 地震震级-频度关系
+
+在地震学中，有一个著名的定律，叫 Gutenberg–Richter 定律（简称 GR law），该定律描述了
+震级与某一地区大于等于该震级的地震数量之间的关系。该定律的表达式是：
+
+$\log_{10} N = a - b M$ 或 $N=10^{a-bM}$
```
Comment (core-man) [inline source/exercises/catalog.md:366]:
```suggestion
震级与某一地区大于等于该震级的地震数量之间的关系。该定律的表达式是：$\log_{10} N = a - b M$。
```

File: source/exercises/catalog.md [hunk 4]
```diff
@@ -94,12 +103,307 @@ USGS 网站提供了在线搜索和查看地震目录的功能，既可以手动
...
+
+其中，
+
+- $M$ 表示震级
+- $N$ 表示震级大于等于 $M$ 的地震数量
+- $a$ 和 $b$ 是常数
```
Comment (core-man) [inline source/exercises/catalog.md:372]:
```suggestion

其中，$M$ 表示震级，$N$ 表示震级大于等于 $M$ 的地震数量，$a$ 和 $b$ 是常数。
```

File: source/exercises/catalog.md [hunk 5]
```diff
@@ -94,12 +103,307 @@ USGS 网站提供了在线搜索和查看地震目录的功能，既可以手动
...
+```
+
+为了进一步对数据做处理以及绘图，我们还需要导入 [NumPy](https://numpy.org/)
+和 [Matplotlib](https://matplotlib.org/) 模块：
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+import numpy as np
+```
 
 ### 地震深度分布直方图
```
Comment (core-man) [inline source/exercises/catalog.md:326]: I feel the following two subsections could be moved to `data analysis` sections. We should focus on catalog downloading in this section, otherwise, it's too long. The readers might lose themselves.

Comment (seisman) [reply source/exercises/catalog.md]: Done.

File: source/exercises/catalog-analysis.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# 分析地震目录
+
+- 本节贡献者: {{田冬冬}}（作者）、{{姚家园}}（审稿）
+- 最近更新日期: 2022-07-30
```
Comment (seisman) [inline source/exercises/catalog-analysis.md:17]:
```suggestion
- 最近更新日期: 2022-07-31
```

File: source/exercises/catalog.md [hunk 7]
```diff
@@ -13,18 +13,27 @@ kernelspec:
 
 # 地震目录
 
-- 本节贡献者: {{田冬冬}}（作者）
-- 最近更新日期: 2022-07-17
-- 预计花费时间: 30 分钟
+- 本节贡献者: {{田冬冬}}（作者）、{{姚家园}}（审稿）
+- 最近更新日期: 2022-07-21
```
Comment (seisman) [inline source/exercises/catalog.md:17]:
```suggestion
- 最近更新日期: 2022-07-31
```

File: source/exercises/catalog.md [hunk 8]
```diff
@@ -13,18 +13,27 @@ kernelspec:
...
+- 预计花费时间: 120 分钟
```
Comment (seisman) [inline source/exercises/catalog.md:18]:
```suggestion
- 预计花费时间: 90 分钟
```

File: source/exercises/catalog-analysis.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+- 本节贡献者: {{田冬冬}}（作者）、{{姚家园}}（审稿）
+- 最近更新日期: 2022-07-31
+- 预计花费时间: 15 分钟
+
+---
+
+在获取地震目录之后，通常还需要对地震目录做一些简单的绘图和分析。这一节演示如何
+绘制最简单的地震深度直方图、地震震级直方图以及震级-频次关系图。
+
+首先，需要把地震目录准备好。这里我们选择下载 2000 - 2010 年间震级大于 4.8 级，
```
Comment (core-man) [inline source/exercises/catalog-analysis.md:25]:
```suggestion
首先，需要把地震目录准备好。这里我们选择下载 2000–2010 年间震级大于 4.8 级，
```

File: source/exercises/catalog-analysis.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+震源深度大于 50 km 的地震。这一地震目录中共计约 7000 个地震：
+```{code-cell} ipython3
+from obspy.clients.fdsn import Client
+
+client = Client("USGS")
```
Comment (core-man) [inline source/exercises/catalog-analysis.md:30]: See https://seismo-learn.org/sitepreview/seismo-learn/seismology101/obspy-get-events/exercises/catalog-analysis/ for an bug.

**Review by core-man — APPROVED** [review]

File: source/exercises/catalog-analysis.md (hunk 10 above)
Comment (seisman) [reply source/exercises/catalog-analysis.md:30]: It's due to temporary network issues so can be safely ignored.
